lib/recent.go

```diff
@@ -7,61 +7,44 @@ import (
 	"strings"
 )
 
-type RecentFiles struct {
+type RecentFile struct {
 	Terraform []string `json:"terraform"`
 	OpenTofu  []string `json:"opentofu"`
 }
 
-func addRecent(requestedVersion string, installPath string, distribution string) {
+func addRecent(requestedVersion string, installPath string, product Product) {
 	if !validVersionFormat(requestedVersion) {
 		logger.Errorf("The version %q is not a valid version string and won't be stored", requestedVersion)
 		return
 	}
 	installLocation := GetInstallLocation(installPath)
 	recentFilePath := filepath.Join(installLocation, recentFile)
-	var recentFileData RecentFiles
+	var recentFileData RecentFile
 	if CheckFileExist(recentFilePath) {
 		unmarshalRecentFileData(recentFilePath, &recentFileData)
 	}
-	prependRecentVersionToList(requestedVersion, distribution, &recentFileData)
+	prependRecentVersionToList(requestedVersion, product, &recentFileData)
 	saveRecentFile(recentFileData, recentFilePath)
 }
 
-func prependRecentVersionToList(version, distribution string, r *RecentFiles) {
-	var sliceToCheck []string
-	switch distribution {
-	case distributionTerraform:
-		sliceToCheck = r.Terraform
-	case distributionOpenTofu:
-		sliceToCheck = r.OpenTofu
-	}
+func prependRecentVersionToList(version string, product Product, r *RecentFile) {
+	sliceToCheck := product.GetRecentVersionProduct(r)
 	for versionIndex, versionValue := range sliceToCheck {
 		if versionValue == version {
 			sliceToCheck = append(sliceToCheck[:versionIndex], sliceToCheck[versionIndex+1:]...)
 		}
 	}
 	sliceToCheck = append([]string{version}, sliceToCheck...)
 
-	switch distribution {
-	case distributionTerraform:
-		r.Terraform = sliceToCheck
-	case distributionOpenTofu:
-		r.OpenTofu = sliceToCheck
-	}
+	product.SetRecentVersionProduct(r, sliceToCheck)
 }
 
-func getRecentVersions(installPath string, dist string) ([]string, error) {
+func getRecentVersions(installPath string, product Product) ([]string, error) {
 	installLocation := GetInstallLocation(installPath)
 	recentFilePath := filepath.Join(installLocation, recentFile)
-	var recentFileData RecentFiles
+	var recentFileData RecentFile
 	unmarshalRecentFileData(recentFilePath, &recentFileData)
```

Comment:
Can you check if the file exist before unmarshalling or reading?
 If a first time user downloads tfswitch, they won't have a RECENT file. They will be greeted with a error message (even though, we will create a RECENT file for them later)
 
 
 
 something like :
 
 
 
 ```
 
 	if CheckFileExist(recentFilePath) {
		unmarshalRecentFileData(recentFilePath, &recentFileData)
	}
	
	```

Reply:
Since we call `unmarshalRecentFileData` in a couple of places, would you be happy to have the check inside `unmarshalRecentFileData`, so it just returns if the file doesn't exist?

Reply:
Have done so in 0d34959, let me know what you think :)